stubs/oauthlib/oauthlib/oauth2/rfc6749/clients/base.pyi

```diff
@@ -46,9 +46,9 @@ class Client:
     ) -> None: ...
     @property
     def token_types(self): ...
-    def prepare_request_uri(self, *args, **kwargs) -> None: ...
-    def prepare_request_body(self, *args, **kwargs) -> None: ...
-    def parse_request_uri_response(self, *args, **kwargs) -> None: ...
+    def prepare_request_uri(self, *args, **kwargs) -> AnyStr: ...
```

Comment:
If I read the implementation correctly, this should actually return just `str`:

```suggestion
    def prepare_request_uri(self, *args, **kwargs) -> str: ...
```

Reply:
Maybe an easier way is to look at examples in the docstrings. I used `grep` to find them:

```
akuli@akuli-desktop:/tmp/e/lib/python3.9/site-packages/oauthlib/oauth2/rfc6749/clients$ grep -A1 prepare_request_uri *
base.py:    def prepare_request_uri(self, *args, **kwargs):
base.py-        """Abstract method used to create request URIs."""
--
base.py:        auth_url = self.prepare_request_uri(
base.py-            authorization_url, redirect_uri=self.redirect_url,
--
mobile_application.py:    def prepare_request_uri(self, uri, redirect_uri=None, scope=None,
mobile_application.py-                            state=None, **kwargs):
--
mobile_application.py:            >>> client.prepare_request_uri('https://example.com')
mobile_application.py-            'https://example.com?client_id=your_id&response_type=token'
mobile_application.py:            >>> client.prepare_request_uri('https://example.com', redirect_uri='https://a.b/callback')
mobile_application.py-            'https://example.com?client_id=your_id&response_type=token&redirect_uri=https%3A%2F%2Fa.b%2Fcallback'
mobile_application.py:            >>> client.prepare_request_uri('https://example.com', scope=['profile', 'pictures'])
mobile_application.py-            'https://example.com?client_id=your_id&response_type=token&scope=profile+pictures'
mobile_application.py:            >>> client.prepare_request_uri('https://example.com', foo='bar')
mobile_application.py-            'https://example.com?client_id=your_id&response_type=token&foo=bar'
grep: __pycache__: On hakemisto
--
web_application.py:    def prepare_request_uri(self, uri, redirect_uri=None, scope=None,
web_application.py-                            state=None, code_challenge=None, code_challenge_method='plain', **kwargs):
--
web_application.py:            >>> client.prepare_request_uri('https://example.com')
web_application.py-            'https://example.com?client_id=your_id&response_type=code'
web_application.py:            >>> client.prepare_request_uri('https://example.com', redirect_uri='https://a.b/callback')
web_application.py-            'https://example.com?client_id=your_id&response_type=code&redirect_uri=https%3A%2F%2Fa.b%2Fcallback'
web_application.py:            >>> client.prepare_request_uri('https://example.com', scope=['profile', 'pictures'])
web_application.py-            'https://example.com?client_id=your_id&response_type=code&scope=profile+pictures'
web_application.py:            >>> client.prepare_request_uri('https://example.com', code_challenge='kjasBS523KdkAILD2k78NdcJSk2k3KHG6')
web_application.py-            'https://example.com?client_id=your_id&response_type=code&code_challenge=kjasBS523KdkAILD2k78NdcJSk2k3KHG6'
web_application.py:            >>> client.prepare_request_uri('https://example.com', code_challenge_method='S256')
web_application.py-            'https://example.com?client_id=your_id&response_type=code&code_challenge_method=S256'
web_application.py:            >>> client.prepare_request_uri('https://example.com', foo='bar')
web_application.py-            'https://example.com?client_id=your_id&response_type=code&foo=bar'
```

If the library returns a `bytes` object instead, it is a bug in the library. Users do not need to handle `bytes` return values.